packages/store/src/database/backends/indexedDB.ts

```diff
@@ -22,6 +22,18 @@ import {
   DexieEventRowWithPrimaryKey,
 } from "./dexie/tables";
 
+function getEntityRowForEntityRecord(
+  record: DatabaseBackendEntityRecord<Entity>,
+) {
```

Comment:
(feel free to ignore): could we type the return to make it clearer when the Dexie record changes in the future.
```suggestion
function getEntityRowForEntityRecord(
  record: DatabaseBackendEntityRecord<Entity>,
): DexieEntityRow {
```

This way the error will point right to this function rather then to the consumer of it.